### ISSUE TITLE
Not working on list view

### ISSUE THREAD

**Issue by mjanos10** [marmelab/aor-permissions#20]
Hi!

Maybe I'm missing something but for me the <WithPermission> is not working correctly on List views.
## First Example

![image](https://user-images.githubusercontent.com/13365853/28289979-f7208ec0-6b44-11e7-80cc-7285868e3639.png)

And here is the code for it:

```js
export const TrainingList = (props) => {
	return <List {...props} perPage={25} filters={<TrainingFilter />}>
		<Datagrid>
			{/* Other stuff */}
			<WithPermission value={["gymadmin", "mainAdmin"]} >
				<AcivateTraining label="Aktíválás" addLabel/>
			</WithPermission>
                       {/* Other stuff */}
		</Datagrid>
	</List>
};
```

When `addLabel` prop was not present the label wasn't even showing up anywhere.

## Second Example

![image](https://user-images.githubusercontent.com/13365853/28290184-ace54444-6b45-11e7-9268-a2f6baed4c80.png)

And the code

```js
export const GymUsersList = (props) => (
	<List {...props} perPage={25}>
		<Datagrid>
			<WithPermission value={"mainAdmin"}>
				<TextField source="gymId" />
			</WithPermission>
                       {/* Other stuff */}
		</Datagrid>
	</List>
);
```

Are these the expected behaviour? 
Thanks in advance!
P.S: in Edit or Create view everything works as expected, this only shows up List view.

**Comment by djhi** [comment]
This will not work for the time being. We are working on a solution but this will probably requires changes in Admin-on-rest that won't land until v1.3.0

**Comment by yozzh** [comment]
Here is temporary workaround for it. Just create simple component and proxy props through <WithPermission>
```
const EditUserButton = (props) => {
  return (
    <WithPermission authClient={authClient} value='cpanel_users.can_edit'>
      <EditButton {...props} />
    </WithPermission>
  )
}

<...>

<Datagrid>
  <TextField source='id' />
  <TextField source='username' />
  <EditUserButton />
</Datagrid>
```

**Comment by djhi** [comment]
We are working on integrating this addon into the core admin-on-rest package. I have proposed a solution for this particular problem here too. 

See https://github.com/marmelab/admin-on-rest/pull/959

**Comment by mjanos10** [comment]
@djhi  Oh my, I can't wait for this to happen :)
@yozzh Thanks man, will try it out! :)

**Comment by smallAttr** [comment]
you can use SwitchPermissions to solve this issue.
eg:
<SwitchPermissions authClient={authClient} {...props}>
        <Permission value="ROLE_ADMIN">
          <EditButton/>
        </Permission>
      </SwitchPermissions>
but it isn't a good way

**Comment by smallAttr** [comment]
you can use SwitchPermissions to solve this issue.
eg:
`<SwitchPermissions authClient={authClient} {...props}>
        <Permission value="ROLE_ADMIN">
          <EditButton/>
        </Permission>
      </SwitchPermissions>`
but it isn't a good way.

**Comment by djhi** [comment]
It has been merged! It will be available in `1.3.0`. Hope you'll enjoy it!

**Comment by djhi** [comment]
**This project is discontinued. Its features have been merged in admin-on-rest 1.3.0. See [admin-on-rest documentation](https://marmelab.com/admin-on-rest/Authorization.html)**
